linode_metadata/metadata_client.py

```diff
@@ -70,6 +78,21 @@ def __init__(
         self.base_url = base_url
         self._append_user_agent = user_agent
         self.timeout = timeout
+        self._debug = debug
+        if debug:
+            self._logger = logging.getLogger("MDS debug")
```

Comment:
nitpick: It might make sense to name the logger something like `linode_metadata` to match the package name 👍